### ISSUE TITLE
Improve search result accessibility

### ISSUE THREAD

**Issue by CMessinides** [CMessinides/a11ypod#16]
After navigating the search results page with a screen reader, I think the accessibility of the search  results could be improved. Here are a few specific opportunities:

- The podcast previews currently render the publisher within a `<dd>` tag, related to the `<dt>` with the content "Publisher," all within a definitions list `<dl>`. However, this markup can be jarring within the context of the search results list, as the screen reader will pause to jump into the `<dl>` and pause again when it jumps out. Here's an example of how it sounds:

  ```
  List item
  Heading level 2
  The Ringer NBA Show
  List with 2 items
  Publisher
  [...wait for user command]
  The Ringer
  [...wait for user command]
  Out of list
  List item... [etc.]
  ```

  A better solution would be to simply wrap the publisher name in a `<span>` tag and label it with some screen-reader only text, so it would be announced like `Publisher: The Ringer` without pausing.

- Updating the alert as the user types causes some lag, like so:
  
  ```
  Loading results for term 'n'
  Loading results for term 'nb'
  Loading results for term 'nba'
  Found 25 results for term 'nba'
  ```
  
  The alert won't just skip to the most recent text content -- it can't "interrupt" itself, so to speak. It could be worth experimenting with `aria-live="assertive"`, but I'm not sure if that will solve the lag issue. More likely, some sort of debouncing will have to be done to ensure only the most relevant and recent alert message gets rendered.

- On the topic of the alert message, the language could be a little more helpful. If more results were loaded, it should say how many were loaded, not just the new total, i.e. `Loaded 25 more results` instead of `Found 50 results for 'nba'`. And every time new results are found or more results are loaded, the alert should announce whether more results can be loaded. Otherwise, people using screen readers will have make an assumption based on the presence/absence of a "Load More Results" button, and that puts uncertainty on the user ("Are there really no more results, or did the button not render? Should I refresh to double-check?", etc.).
